With Python 3.11 and later, mercurial can fail with recursion depth
exceeded error when one of the extensions includes lxml.

This only happens when lxml.etree is lazy loaded, so we include it on
the list of modules that are loaded immediately.

This was reported upstream:
https://bz.mercurial-scm.org/show_bug.cgi?id=6932

--- mercurial-7.1.2/hgdemandimport/__init__.py
+++ mercurial-7.1.2/hgdemandimport/__init__.py
@@ -65,6 +65,8 @@ IGNORES = {
     'warnings',
     'threading',
     'collections.abc',
+    # not compatible with on-demand loading
+    'lxml.etree',
 }
 
 _pypy = '__pypy__' in sys.builtin_module_names
